### PR TITLE
[WIP] Add kubeconfig certificate rotation tests

### LINKED ISSUE #520: Kubeconfig client certs not automatically renewed
**What happened:**
The client cert in the generated kubeconfig secret expires after one year. However, the rke2 provider does not regenerate the kubeconfig with a new cert, breaking any tools that depend on reading the clustername-kubeconfig secret to get cluster credentials.

**What did you expect to happen:**

The kubeconfig secret is renewed a reasonable time before expiration. The cluster-api kubeadm control plane does this [here](https://github.com/kubernetes-sigs/cluster-api/blob/92e16dbdaead4dd3e1f9ebec2c0d395733d63d5f/controlplane/kubeadm/internal/controllers/helpers.go#L98)

**How to reproduce it:**

Create a cluster. Wait one year. Check kubeconfig cert expiration

**Anything else you would like to add:**

Weirdly the code has a [comment](https://github.com/rancher/cluster-api-provider-rke2/blob/053adabd649465984799af3408575762fa9c78d5/controlplane/internal/controllers/rke2controlplane_controller.go#L810) "only do rotation on owned secrets", checks that it owns the kubeconfig secret, and then just doesn't rotate it. 

**Environment:**

- rke provider version: 0.8.0
- OS (e.g. from `/etc/os-release`): ubuntu 22.04 LTS

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
Adds new test cases for kubeconfig certificate automatic rotation implementation in https://github.com/rancher/cluster-api-provider-rke2/pull/521

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #520

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits into logical changes
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Review by yiannistri — APPROVED** [review]

**Review by anmazzotti — APPROVED** [review]

**Review by alexander-demicev — CHANGES_REQUESTED** [review]
adding "requesting changes" so the PR doesn't get merged and close the issue. @furkatgofurov7 would you be able to verify that rotating kubeconfig causes an update for machines? Maybe trigger it somehow manually from code, I don't have ideas how to verify it in other ways at the moment

File: controlplane/internal/controllers/rke2controlplane_controller_test.go [hunk 1]
```diff
@@ -251,4 +258,129 @@ var _ = Describe("Reconcile control plane conditions", func() {
...
+		// Check that rotation is needed
+		needsRotation, err := kubeconfig.NeedsClientCertRotation(secret, certs.ClientCertificateRenewalDuration)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(needsRotation).To(BeTrue())
+
+		// Rotate kubeconfig secret
+		_, err = r.reconcileKubeconfig(ctx, clusterName, endpoint, rcp)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(testEnv.Get(ctx, client.ObjectKey{Namespace: ns.Name, Name: secret.Name}, secret)).To(Succeed())
```
Comment (Danil-Grigorev) [inline controlplane/internal/controllers/rke2controlplane_controller_test.go:287]: Can we also check that the kubeconfig secret is actually rotated? It seems to me that the current validation equals to “no rotation” test case: https://github.com/rancher/cluster-api-provider-rke2/pull/550/files#diff-cd4559684c03b3b2a238ed4f8c142bb867312c79174f5a6474b157ac7849a737R312-R316

**Comment by furkatgofurov7** [comment]
@alexander-demicev @Danil-Grigorev I checked the cert rotation code again and found out related tests in the kubeadm code so I think: we should not need to test cert rotation (or basically duplicate the tests from kubeadm), since anyways we are just using the function by importing and there is already a comprehensive [tests](https://github.com/kubernetes-sigs/cluster-api/blob/main/util/kubeconfig/kubeconfig_test.go#L405-L436) for it in the kubeadm codebase.

**Comment by alexander-demicev** [comment]
@furkatgofurov7 do we set this annotation on the bootstrap config https://cluster-api.sigs.k8s.io/tasks/certs/auto-rotate-certificates-in-kcp.html?highlight=rotat#triggering-machine-rollout-for-certificate-expiry? and does CAPRKE2 replace machines when it's set and expires?

File: controlplane/internal/controllers/rke2controlplane_controller_test.go (hunk 1 above)
Comment (alexander-demicev) [reply controlplane/internal/controllers/rke2controlplane_controller_test.go:287]: `reconcileKubeconfig` calls  `kubeconfig.CreateSecretWithOwner`, I'm also not sure that cert would get rotated.
